### PR TITLE
Bot docs

### PR DESCRIPTION
**Pull request by joshblum**

### CODE REVIEW

**Comment by joshblum** [comment]
Very raw, needs proofreading + polishing :)

File: docs/source/bots.rst [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+
+Below, we'll walk through the various bots available in Orchestra.
+
+*********
+StaffBot
+*********
+
+``StaffBot`` provides a simple way to ask a group of workers if they would like
+to work on a particular ``Task``. The goal is reduce the human load when
+finding a ``Worker`` for  a ``Task``. ``StaffBot`` allows a user to interact
```
Comment (marcua) [inline docs/source/bots.rst]: ``Task.``

File: docs/source/bots.rst [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+with Orchestra via `Slack`_ or can be configured to automatically find workers for
+a ``Task`` when it is available by setting up an `Assignment Policy`_.
+
+``StaffBot`` works by reaching out to qualified workers and offering them a
+``Task`` to work on. Workers can then either accept or reject the task and
```
Comment (marcua) [inline docs/source/bots.rst]: accept or reject the task (via Slack or email)

File: docs/source/bots.rst [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+            'allowed_user_ids': ['allowed_user_ids'],
+            'allowed_user_names': ['allowed_user_names'],
+            'allowed_commands': ['allowed_commands'],
+        }
+
+``StaffBot`` will use the ``STAFFBOT_CONFIG`` to filter messages that do not
+match items in the list. If a parameter is not specified, all values are
+accepted by default.
+
+
```
Comment (marcua) [inline docs/source/bots.rst]: delete newline?

File: docs/source/bots.rst [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+Once this configuration is complete, you can test by typing ``/staffbot staff
```
Comment (marcua) [inline docs/source/bots.rst]: test it

File: docs/source/bots.rst [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+<task-id>`` where ``<task-id>`` is an unassigned task id.
```
Comment (marcua) [inline docs/source/bots.rst]: "By default, StaffBot will reach out to `Workers` who have the appropriate `WorkerCertification` for the task you are staffing, in batches of X `Workers` every Y minutes.

File: docs/source/bots.rst [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+
+Assignment Policy
+================
+
+``StaffBot`` can also work without user interaction by specifying an Assignment
```
Comment (marcua) [inline docs/source/bots.rst]: "can also automatically staff projects by specifying..."

File: docs/source/bots.rst [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+Policy. Orchestra supports custom logic for assigning workers to tasks, and
+``StaffBot`` leverages this by asking qualified workers if they would like to
+work on a ``Task`` as soon as the ``Task`` is available. To specify the
+``StaffBot`` auto assignment policy, add the following to the ``Step``
```
Comment (marcua) [inline docs/source/bots.rst]: "auto-assignment policy, which uses the same logic as the `/staffbot staff...` command, add the following"

File: docs/source/bots.rst [hunk 8]
```diff
@@ -0,0 +1,114 @@
...
+              "path": "orchestra.bots.assignment_policies.staffbot_autoassign"
+          }
+      }
+  },
+  [...step definition...]
+
+Detailed Description Function
+=============================
+
+The ``detailed_scription_function`` is used to dynamically describe a ``Task``
```
Comment (marcua) [inline docs/source/bots.rst]: spelling: `detailed_description_function`

File: docs/source/bots.rst [hunk 9]
```diff
@@ -0,0 +1,114 @@
...
+when ``StaffBot`` makes requests to workers, offering them the opportunity to
+work on the ``Task``. The function is given a ``task_details`` dictionary and
+can be passed extra ``kwargs`` as shown below::
+
+  [...step definition...]
+  "detailed_description_function": {
+       "path": "orchestra.tests.helpers.fixtures.get_detailed_description",
```
Comment (marcua) [inline docs/source/bots.rst]: show an example `get_detailed_description`

File: docs/source/bots.rst (hunk 2 above)
Comment (joshblum) [inline docs/source/bots.rst]: done

File: docs/source/bots.rst (hunk 3 above)
Comment (joshblum) [inline docs/source/bots.rst]: done

File: docs/source/bots.rst (hunk 4 above)
Comment (joshblum) [inline docs/source/bots.rst]: done

File: docs/source/bots.rst (hunk 5 above)
Comment (joshblum) [inline docs/source/bots.rst]: done

File: docs/source/bots.rst (hunk 6 above)
Comment (joshblum) [inline docs/source/bots.rst]: done

File: docs/source/bots.rst (hunk 7 above)
Comment (joshblum) [inline docs/source/bots.rst]: done

File: docs/source/bots.rst (hunk 8 above)
Comment (joshblum) [inline docs/source/bots.rst]: done

File: docs/source/bots.rst (hunk 9 above)
Comment (joshblum) [inline docs/source/bots.rst]: done

File: docs/source/bots.rst [hunk 10]
```diff
@@ -0,0 +1,124 @@
...
+       }
+  }
+  [...step definition...]
+
+::
+
+  # my_project/orchestra_helpers.py
+  def get_detailed_description(task_details **kwargs):
+    return '''A new task is available!
+              Find out more about {} at example.com/projects/!'''.format(kwargs.get('text'))
```
Comment (marcua) [inline docs/source/bots.rst]: example.com/projects/{} where {} is `task_details['project']['id']`

**Comment by marcua** [comment]
one comment and then lgtm!

**Pull request merged**